zio-http/jvm/src/main/scala/zio/http/netty/NettyBody.scala

```diff
@@ -113,21 +132,26 @@ object NettyBody extends BodyEncoding {
     override private[zio] def unsafeAsArray(implicit unsafe: Unsafe): Array[Byte] =
       ByteBufUtil.getBytes(byteBuf)
 
-    override def contentType(newMediaType: MediaType): Body = copy(mediaType = Some(newMediaType))
+    override def contentType(newMediaType: MediaType): Body = copy(mediaType0 = () => Some(newMediaType))
 
     override def contentType(newMediaType: MediaType, newBoundary: Boundary): Body =
-      copy(mediaType = Some(newMediaType), boundary = boundary.orElse(Some(newBoundary)))
+      copy(mediaType0 = () => Some(newMediaType), boundary0 = () => boundary.orElse(Some(newBoundary)))
```

Comment:
By the way, is this the correct behaviour? I'm a bit confused on why we would keep the old `boundary` if it was defined. Shouldn't this be `boundary = Some(newBoundary)` instead?

Reply:
Probably so